paddle/fluid/imperative/reducer.h

```diff
@@ -188,14 +196,22 @@ class Reducer {
   std::vector<size_t> unused_vars_;
   bool has_marked_unused_vars_{false};
   bool find_unused_vars_{false};
-  bool all_group_ready_{false};
+  bool groups_need_finalize_{false};
 #ifdef PADDLE_WITH_XPU_BKCL
   // comm_pool_ is used for scheduling allreduce in multi Kunlun cards training.
   std::unique_ptr<::ThreadPool> comm_pool_{nullptr};
   uint32_t comm_op_count_;
   std::mutex mutex_;
   std::condition_variable cv_;
 #endif
+
+  // it just for checking hook, each parameter can only trigger one hook
+  std::vector<bool> vars_marked_ready_;
+
+  // Following variables are to help control flow,
+  std::vector<int> local_used_vars_;
```

Comment:
Could we find better name, or add more comment to explain this vector.
Because after all-reduce, what it denotes is global_used_vars instead of local_used_vars

Reply:
done